app/src/main/java/org/thoughtcrime/securesms/net/SocksProxy.java

```diff
@@ -49,7 +49,7 @@ public String getUrl() {
   public InetSocketAddress getSocketAddress() {
     InetSocketAddress address = null;
     try {
-      address = new InetSocketAddress(host, port);
+      address = InetSocketAddress.createUnresolved(host, port);
```

Comment:
I'm not sure, but I'd say the address needs to be resolved at this point. Could you try setting a non-trivial hostname (i.e. fully qualified DNS name, not localhost) and test if it still works?

Reply:
Tested and it works! I also fixes a potential ANR error.